### PR TITLE
Refactor fuse sidecar container injection

### LINKED ISSUE #2148: [FEATURES]Refactor fuse sidecar container injection
**What feature you'd like to add:**

**Why is this feature needed:**

### PR DESCRIPTION
**Pull request by TrafalgarZZZ**
### Ⅰ. Describe what this PR does
The original fuse sidecar mutation logic can be described as follow:
FuseSidecar.Mutate() -> InjectPod() -> Inject() -> inject()

In the inject() func, it takes the following mutation steps:
1. Transform the input object into a FluidApplication object, which contains one or more PodSpecs to be mutated.
2. For each PodSpec in the FluidApplication object, 
   a. check if needs injection by checking labels on the PodSpec (i.e. serverlessEnabled or injectionDone label)
   b. Generate fuse container template to inject.
   c. Handle mutations on the PodSpec's volumes (including modifying existing volumes and append new volumes)
   d. Handle mutations on the PodSpec's containers (including mutating container's volumeMounts and appending fuse container)
   e. Add injectionDone label to the PodSpec
3. return the mutated FluidApplication's object.

In the Inject() func, it for-loop iterates `runtimeInfos`, and call `inject()` func for each `pvcName` and `runtimeInfo` until all the runtimeInfo has been used to mutate the Pod.

**This PR does the following refactoring:**
- Move the for-loop iteration on `runtimeInfos` to `inject()` func after Step 1 but before Step 2.
- Extract utility functions for sub-steps in Step 2 (e.g. `appendVolumes()`, `mutateVolumeMounts()`) to improve readability
- Break Step 2.d into two utility functions `mutateVolumeMounts()` and `injectFuseContainerToFirst()`
- Add injectionDone label immediately **after the for-loop iteration on `runtimeInfos` finished** for each PodSpec

### Ⅱ. Does this pull request fix one issue?

fixes #2148 

### Ⅴ. Special notes for reviews

### CODE REVIEW

File: pkg/application/inject/fuse/container_injector.go [hunk 1]
```diff
@@ -17,84 +17,55 @@ limitations under the License.
 package fuse
 
 import (
-	"fmt"
 	"github.com/fluid-cloudnative/fluid/pkg/common"
-	"github.com/fluid-cloudnative/fluid/pkg/scripts/poststart"
-	"github.com/fluid-cloudnative/fluid/pkg/utils"
 	corev1 "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/types"
 )
```
Comment (cheyang) [inline pkg/application/inject/fuse/container_injector.go:23]: How about adding comments to the function name to improve the understanding of the logic?

Comment (TrafalgarZZZ) [reply pkg/application/inject/fuse/container_injector.go:23]: Done

File: pkg/application/inject/fuse/injector.go [hunk 2]
```diff
@@ -144,221 +149,188 @@ func (s *Injector) inject(in runtime.Object, pvcName string, runtimeInfo base.Ru
...
 	pods, err := application.GetPodSpecs()
 	if err != nil {
 		return
 	}
 
 	for _, pod := range pods {
-		metaObj, err := pod.GetMetaObject()
-		if err != nil {
-			return out, err
+		for pvcName, runtimeInfo := range runtimeInfos {
```
Comment (cheyang) [inline pkg/application/inject/fuse/injector.go:158]: Is it better to check labels before injectObject?

```golang
if !utils.ServerlessEnabled(metaObj.Labels) || utils.InjectSidecarDone(metaObj.Labels) {
		log.V(1).Info("Serverless injection not enabled in pod labels, skip",
			"name", namespacedName.Name,
			"namespace", namespacedName.Namespace)
		return nil
	}
```

Comment (TrafalgarZZZ) [reply pkg/application/inject/fuse/injector.go:167]: Done

File: pkg/application/inject/fuse/injector.go [hunk 3]
```diff
@@ -144,221 +149,175 @@ func (s *Injector) inject(in runtime.Object, pvcName string, runtimeInfo base.Ru
...
 	log.V(1).Info("Inject application", "namespacedName", namespacedName, "kind", kind)
-
 	pods, err := application.GetPodSpecs()
 	if err != nil {
 		return
 	}
 
 	for _, pod := range pods {
-		metaObj, err := pod.GetMetaObject()
+		shouldSkipInjection, err := s.shouldSkipInjection(pod, namespacedName)
```
Comment (cheyang) [inline pkg/application/inject/fuse/injector.go:158]: I suggest naming the method name as `shouldInjection`.

File: pkg/application/inject/fuse/injector.go [hunk 4]
```diff
@@ -144,221 +149,175 @@ func (s *Injector) inject(in runtime.Object, pvcName string, runtimeInfo base.Ru
...
+			"name", namespacedName.Name,
+			"namespace", namespacedName.Namespace,
+			"pvc", pvcName,
+			"mounted pvcs", mountedPvc)
+		return nil
+	}
 
-			volumes = append(volumes, appScriptGen.GetVolume())
-		}
+	// 2. generate fuse container template for injection.
```
Comment (cheyang) [inline pkg/application/inject/fuse/injector.go:218]: remove the line between comments and source code.

File: pkg/application/inject/fuse/injector.go [hunk 5]
```diff
@@ -144,221 +149,175 @@ func (s *Injector) inject(in runtime.Object, pvcName string, runtimeInfo base.Ru
...
-			log.V(1).Info("after injection",
-				"podName", namespacedName,
-				"pvcName", pvcName,
-				"containers", containers)
-			err = pod.SetContainers(containers)
-			if err != nil {
-				return out, err
-			}
-		}
+	var (
```
Comment (cheyang) [inline pkg/application/inject/fuse/injector.go:230]: move the variables definition to the start of the function.

File: pkg/application/inject/fuse/injector.go (hunk 3 above)
Comment (TrafalgarZZZ) [reply pkg/application/inject/fuse/injector.go]: renamedit as `shouldInject`

**Review by cheyang — APPROVED** [review]

**Pull request merged**